sound/soc/sof/ipc3-dtrace.c

```diff
@@ -419,13 +428,16 @@ static int ipc3_dtrace_enable(struct snd_sof_dev *sdev)
 	dev_dbg(sdev->dev, "%s: stream_tag: %d\n", __func__, params.stream_tag);
 
 	/* send IPC to the DSP */
+	sdev->dtrace_state = SOF_DTRACE_INITIALIZING;
```

Comment:
I don't fully get the proposal and what happens if the IPC fails. we would still allow the client to read stuff from the buffer?

Reply:
@plbossart, I see. I will set the state to SOF_DTRACE_DISABLED if either the ipc or the host operation fails.